packages/react-components/react-dialog/src/components/DialogTitle/useDialogTitle.tsx

```diff
@@ -17,21 +18,21 @@ import { useDialogTitleInternalStyles } from './useDialogTitleStyles.styles';
  * @param ref - reference to root HTMLElement of DialogTitle
  */
 export const useDialogTitle_unstable = (props: DialogTitleProps, ref: React.Ref<HTMLElement>): DialogTitleState => {
-  const { as, action } = props;
+  const { as = defaultComponentType, action } = props;
   const modalType = useDialogContext_unstable(ctx => ctx.modalType);
   const internalStyles = useDialogTitleInternalStyles();
 
   return {
-    components: {
-      root: 'h2',
-      action: 'div',
-    },
-    root: getNativeElementProps(as ?? 'h2', {
-      ref,
-      id: useDialogContext_unstable(ctx => ctx.dialogTitleId),
-      ...props,
-    }),
-    action: resolveShorthand(action, {
+    root: slot<DialogTitleProps>(
```

Comment:
Why do you need to specify `DialogTitleProps` explicitly?

Reply:
You don't actually. if you remove it'll work. I guess it's there for some implementation reminiscent